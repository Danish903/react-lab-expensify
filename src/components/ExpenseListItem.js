import React from 'react';

const Item = ({ description, amount, createdAt }) => (
  <div>
    <h4>{description}</h4>
    <p>
      ${amount} - {createdAt}
    </p>
  </div>
);

export default Item;
