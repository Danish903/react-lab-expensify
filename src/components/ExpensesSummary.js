import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => (
  <div className="page-header">
    <div className="content-container">
      <h1 className="page-header__title">
        Viewing <span>{expensesCount}</span>{' '}
        {expensesCount > 1 ? 'expenses' : 'expense'} totalling{' '}
        <span>{numeral(expensesTotal / 100).format('$0,0.00')}</span>
      </h1>
      <div className="page-header__actions">
        <Link className="button" to="/create">
          Add Expenses
        </Link>
      </div>
    </div>
  </div>
);
const mapStateToProps = state => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
