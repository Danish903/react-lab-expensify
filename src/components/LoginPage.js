import React from 'react';
import { connect } from 'react-redux';
import { startGoogleLogin, startGithubLogin } from '../actions/auth';

export const LoginPage = props => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <p>Get your expenses under control!</p>
      <div>
        <button
          className="button button--login"
          onClick={props.startGoogleLogin}
        >
          Login with Google
        </button>
        <button
          className="button button--login"
          onClick={props.startGithubLogin}
        >
          Login with Github
        </button>
      </div>
    </div>
  </div>
);

const mapDispatchToProps = dispatch => ({
  startGoogleLogin: () => dispatch(startGoogleLogin()),
  startGithubLogin: () => dispatch(startGithubLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
