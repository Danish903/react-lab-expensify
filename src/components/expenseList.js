import React from 'react';

const ExpenseList = () => (
  <div>
    <h3> Expense List</h3>
  </div>
);

export default ExpenseList;
