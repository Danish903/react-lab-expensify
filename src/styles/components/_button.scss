.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888888;
}

.button--login {
  margin:0 0 $s-size 0; 
  flex:1;
}
