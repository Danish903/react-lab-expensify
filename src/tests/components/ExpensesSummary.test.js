import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixures/expenses';

test('should render summary with single expense correctly', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[expenses[1]]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render summary with plural expenses correctly', () => {
  const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});
