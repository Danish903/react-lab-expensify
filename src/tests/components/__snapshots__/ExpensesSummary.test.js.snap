// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render summary with plural expenses correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        12
      </span>
       
      expenses
       totalling
       
      <span>
        $32.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expenses
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render summary with single expense correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling
       
      <span>
        $3.20
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expenses
      </Link>
    </div>
  </div>
</div>
`;
